mteb/models/model_implementations/eagerworks_models.py

```diff
@@ -0,0 +1,334 @@
+from mteb.models.models_protocols import PromptType
+from mteb.types import Array, BatchedInput
+
+
+class EagerEmbedV1Wrapper(AbsEncoder):
+    """Wrapper for EagerEmbed single-vector embedding models."""
+
+    def __init__(
+        self,
+        model_name: str,
+        revision: str | None = None,
+        device: str | None = None,
+        image_size: int = 784,
+        use_peft: bool = False,
+        **kwargs,
+    ):
+        requires_image_dependencies()
+        requires_package(
+            self, "peft", model_name, "pip install mteb[peft]"
+        )
+        requires_package(
+            self, "qwen_vl_utils", model_name, "pip install mteb[qwen_vl_utils]"
+        )
+        from transformers import AutoProcessor, Qwen3VLForConditionalGeneration
+
+        self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+        self.image_size = image_size
+        self.use_peft = use_peft
+
+        # Handle deprecated torch_dtype parameter
+        if 'torch_dtype' in kwargs:
+            kwargs['dtype'] = kwargs.pop('torch_dtype')
+
+        # Load model
+        if self.use_peft:
+            from peft import PeftModel, PeftConfig
+            config = PeftConfig.from_pretrained(model_name)
+            base_model = Qwen3VLForConditionalGeneration.from_pretrained(
+                config.base_model_name_or_path,
+                **kwargs
+            )
+            self.mdl = PeftModel.from_pretrained(base_model, model_name)
+            self.mdl = self.mdl.merge_and_unload()
+        else:
+            self.mdl = Qwen3VLForConditionalGeneration.from_pretrained(
+                model_name,
+                **kwargs
+            )
+
+        self.mdl = self.mdl.to(self.device)
+        self.mdl.eval()
+
+        # Load processor
+        self.processor = AutoProcessor.from_pretrained(model_name)
+        self.processor.tokenizer.padding_side = "left"
+
+    def get_embedding(self, last_hidden_state: torch.Tensor) -> torch.Tensor:
+        """Extract embeddings from last token of last hidden state."""
+        reps = last_hidden_state[:, -1]
+        return reps
+
+    def encode(
+        self,
+        inputs: DataLoader[BatchedInput],
+        *,
+        task_metadata: TaskMetadata,
+        hf_split: str,
+        hf_subset: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> Array:
+        text_embeddings = None
+        image_embeddings = None
+
+        if "text" in inputs.dataset.features:
+            text_embeddings = self.get_text_embeddings(inputs, **kwargs)
+
+        if "image" in inputs.dataset.features:
+            image_embeddings = self.get_image_embeddings(inputs, **kwargs)
+
+        if text_embeddings is not None and image_embeddings is not None:
+            if len(text_embeddings) != len(image_embeddings):
+                raise ValueError(
+                    "The number of texts and images must have the same length"
+                )
```

Comment:
this is not a requirement on our end (one text can contain multiple images) - if you want to keep this then I would change the message to specifically refer to the (this model does not support ... or this implementation of the model does not suport) to avoid confusion.

Reply:
I took inspiration from here https://github.com/embeddings-benchmark/mteb/blob/main/mteb/models/model_implementations/jina_models.py#L329
but changed the message to what you suggested